src/elements/Svg.tsx

```diff
@@ -45,6 +45,40 @@ export interface SvgProps extends GProps, ViewProps, HitSlop {
   title?: string;
 }
 
+export type DataUrlOptions = JpegOptions | PngOptions;
+
+interface JpegOptions {
+  format: 'jpeg';
+  quality?: number;
+  width: NumberProp;
+  height: NumberProp;
+}
+
+interface PngOptions {
+  format: 'png';
+  width: NumberProp;
+  height: NumberProp;
+}
```

Comment:
`width` and `height` should not be mandatory. User should be able to use 
```ts
ref.toDataURL(someCallback, { format: 'jpeg', quality: 1 });
```
without specifying the width and height

Reply:
In our [implementation](https://github.com/software-mansion/react-native-svg/pull/2431/files#diff-767ae05982e97c6a98efdf996bcad641585b80c35046ecf9c96c77c3dc7d146bL45), an error occurs when you provide any options but don't provide width and height.
Or should we change the whole implementation?

Reply:
Yes, the current implementation allows using `toDataURL(() => {});` without specifying the size, so we shouldn't require specifying it just because someone want to change the format.

Reply:
I know it can be challenging to handle the types correctly (e.g. making `width` required when `height` is specified and vice versa), so we could deprecate the old options for now and introduce a new `size` property.

Reply:
@jakex7 Okay, I thought maybe there would be significant changes and decided not to do that :)
But I'll provide those changes ;)